outbound/state/tests/test_outbound_state.py

```diff
@@ -0,0 +1,169 @@
+                            ReferenceTransactionType(ReferenceTransactionType.TransactionType.ACCEPTANCE)]
+                state.create_new_outbound_state(segments, OPERATION_ID)
+        with self.subTest('None timestamp'):
+            with self.assertRaises(ValueError):
+                segments = [InterchangeHeader(sender='test_sender',
+                                              recipient='test_recipient',
+                                              date_time=None,
+                                              sequence_number=2),
+                            MessageHeader(sequence_number=3),
+                            MessageHeader(sequence_number=7),
+                            ReferenceTransactionNumber(reference=1),
+                            ReferenceTransactionType(ReferenceTransactionType.TransactionType.ACCEPTANCE)]
+                state.create_new_outbound_state(segments, OPERATION_ID)
+        with self.subTest('Empty timestamp string'):
+            with self.assertRaises(ValueError):
+                segments = [InterchangeHeader(sender='test_sender',
+                                              recipient='test_recipient',
+                                              date_time='',
+                                              sequence_number=2),
+                            MessageHeader(sequence_number=3),
+                            MessageHeader(sequence_number=7),
+                            ReferenceTransactionNumber(reference=1),
+                            ReferenceTransactionType(ReferenceTransactionType.TransactionType.ACCEPTANCE)]
+                state.create_new_outbound_state(segments, OPERATION_ID)
+        with self.subTest('None transaction_type'):
+            with self.assertRaises(AttributeError):
+                segments = [InterchangeHeader(sender='test_sender',
+                                              recipient='test_recipient',
+                                              date_time='12:00',
+                                              sequence_number=2),
+                            MessageHeader(sequence_number=3),
+                            MessageHeader(sequence_number=7),
+                            ReferenceTransactionNumber(reference=1),
+                            ReferenceTransactionType(None)]
+                state.create_new_outbound_state(segments, OPERATION_ID)
+        with self.subTest('Not existing transaction_type'):
+            with self.assertRaises(AttributeError):
+                segments = [InterchangeHeader(sender='test_sender',
+                                              recipient='test_recipient',
+                                              date_time='12:00',
+                                              sequence_number=2),
+                            MessageHeader(sequence_number=3),
+                            MessageHeader(sequence_number=7),
+                            ReferenceTransactionNumber(reference=1),
```

Comment:
In IDE warning is raised, Expected type 'str', got 'int' instead. Parameter is expecting int to be passed through.

Reply:
it does not impact the test result, but I can wrap it into the datetime object

Reply:
I think that is should stay as number here, better to improve that __init__ method of message.py:
class ReferenceTransactionNumber(Reference):

    def __init__(self, reference=''):
        super().__init__(qualifier='TN', reference=reference)

Reply:
No, it causes other issue. I would leave it like that. @mattd-kainos, what do you think about it?

Reply:
You're right @Alicja-Kainos the type parameters for Reference and ReferenceTransactionNumber are not quite right. For Reference I'd change it to (None, int, str) and ReferenceTransactionNumber change to (None, int)